File: android/src/main/kotlin/cz/appsdevteam/gptom/mappers/BatchMapper.kt
package cz.appsdevteam.gptom.mappers

import cn.nexgo.smartconnect.model.TransactionResultV2Entity
import cz.appsdevteam.gptom.support.DateUtils
import cz.appsdevteam.gptom.core.JsonKeys

object BatchMapper {

    fun toMap(e: TransactionResultV2Entity): Map<String, Any?> = mapOf(
        JsonKeys.transactionId to e.transactionID,
        
        JsonKeys.batchNumber to e.batchNumber,
        JsonKeys.currencyCode to e.currencyCode?.toString(),

        JsonKeys.date to DateUtils.toISO8601(e.date, e.time),

        JsonKeys.totalCount to e.batchTotalNum,
        JsonKeys.totalAmount to e.batchTotalAmount,
        JsonKeys.saleCount to e.batchSaleNum,
        JsonKeys.saleAmount to e.batchSaleAmount,
        JsonKeys.voidCount to e.batchVoidNum,
        JsonKeys.voidAmount to e.batchVoidAmount,
        
        JsonKeys.tipAmount to e.tipAmount,
    )
}

File: ios/Classes/mappers/BatchMapper.swift
import Foundation

struct BatchMapper {
    private init() {}
    
    static func toMap(_ batch: Batch) -> [String: Any] {
        var b = JsonMapBuilder()
        
        b.put(JsonKeys.amsId, batch.amsId)
        b.put(JsonKeys.batchNumber, batch.batchNumber)
        b.put(JsonKeys.communicationId, batch.communicationId)
        
        b.put(JsonKeys.currencyCode, batch.currency)
        
        b.put(JsonKeys.date, batch.date?.ISO8601Format())
        b.put(JsonKeys.firstTransactionDate, batch.firstTransactionDate?.ISO8601Format())
        b.put(JsonKeys.previousBatchDate, batch.previousBatchDate?.ISO8601Format())
        
        b.put(JsonKeys.saleAmount, batch.saleAmount?.int64)
        b.put(JsonKeys.saleCount, batch.saleCount)
        
        b.put(JsonKeys.totalAmount, batch.totalAmount?.int64)
        b.put(JsonKeys.totalCount, batch.totalCount)
        
        b.put(JsonKeys.voidAmount, batch.voidAmount?.int64)
        b.put(JsonKeys.voidCount, batch.voidCount)
        
        b.put(JsonKeys.invalidCount, batch.invalidCount)
        
        b.put(JsonKeys.tipAmount, batch.tipAmount?.int64)
        b.put(JsonKeys.tipCount, batch.tipCount)
        b.put(JsonKeys.tipAverage, batch.tipAverage?.int64)
        b.put(JsonKeys.tipAveragePercentage, batch.tipAveragePercentage)
        
        return b.build()
    }
}
